Assets/FPS/Scripts/Gameplay/Nautica/BotKillerAgent.cs

```diff
@@ -69,187 +70,203 @@ public override void OnEpisodeBegin()
 			enemies = trainingLevelManager.enemies;
 			pickups = trainingLevelManager.pickups;
 
-            trackedHealth = 1f;
+			trackedHealth = 1f;
 			trackedEnemyHealth.Clear();
 			foreach (var e in enemies) trackedEnemyHealth.Add(1f);
+		}
+
+		private void DebugLogReward(float reward, string rewardtype)
+		{
+			string rewardOutput = "<b>Reward:</b> ";
+			if (reward < 0) rewardOutput = "<color=red>";
+			if (reward > 0) rewardOutput = "<color=green>";
+			rewardOutput += rewardtype + " <b>" + reward.ToString() + "</b></color>, STEP: " + StepCount.ToString() + " Cumulative: " + GetCumulativeReward().ToString();
+			Debug.unityLogger.Log(LOGTAG, rewardOutput);
+		}
+
+		public override void CollectObservations(VectorSensor sensor)
+		{
+			ObserveHealth(sensor);
+			ObserveAmmo(sensor);
+			ObserveEnemies(sensor);
+			ObservePickups(sensor);
+		}
+
+		private void ObserveHealth(VectorSensor sensor)
+		{
+			var healthComponent = GetComponent<Unity.FPS.Game.Health>();
```

Comment:
since we have the `using Unity.FPS.Game` we can should be able to call `GetComponent<Health>()` instead of the fully qualified name.  Doesn't hurt to be specific, but you can also save some extra typing.  Can go either way

Reply:
Replaced 😋